src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1355,19 +1355,24 @@ private void CheckDisallowedAttributeDependentType(TypeWithAnnotations typeArgum
         {
             typeArgument.VisitType(type: null, static (typeWithAnnotations, arg, _) =>
             {
-                var (topLevelType, errorLocation, diagnostics) = arg;
+                var (errorLocation, diagnostics) = arg;
                 var type = typeWithAnnotations.Type;
                 if (type.IsDynamic()
                     || (typeWithAnnotations.NullableAnnotation.IsAnnotated() && !type.IsValueType)
                     || type.IsNativeIntegerWrapperType
                     || (type.IsTupleType && !type.TupleElementNames.IsDefault))
                 {
-                    diagnostics.Add(ErrorCode.ERR_AttrDependentTypeNotAllowed, errorLocation, topLevelType.ToDisplayString(SymbolDisplayFormat.CSharpShortErrorMessageFormat));
+                    diagnostics.Add(ErrorCode.ERR_AttrDependentTypeNotAllowed, errorLocation, type.ToDisplayString(SymbolDisplayFormat.CSharpShortErrorMessageFormat));
```

Comment:
>type.ToDisplayString(SymbolDisplayFormat.CSharpShortErrorMessageFormat) [](http://example.com/codeflow?start=94&length=71)

This is a "preexisting condition", but since we are changing this function and since you are "copying" this below, let's fix it. We are not supposed to convert symbols to strings for the purpose of creating diagnostics. We either should use "raw" symbols as arguments, or, if default formatting isn't what we want, we should use ```FormattedSymbol```s.

Reply:
Makes sense. Fixed: 2fd652d9257dfcb245a738983bdf8364e769c0c2